### ISSUE TITLE
Serialize and deserialize private keys

### ISSUE THREAD

**Issue by shader** [greglook/clj-pgp#16]
The core/encode-ascii function generates a "PGP MESSAGE" for a private key from a generated keypair, but there is no corresponding decoding operation.

For that matter, simply taking the raw bytes from core/encode and then attempting to decode them also does not work.

What is the correct way to serialize and deserialize a private key?

**Comment by greglook** [comment]
I think this reproduces the issue, can you confirm?

```clojure
=> (require '(clj-pgp [core :as pgp] [generate :as gen] [keyring :as keyring]))

=> (gen/generate-keys "test" "s3cr3t" (master-key (keypair (gen/rsa-keypair-generator 2048) :rsa-general)))
{:public #<org.bouncycastle.openpgp.PGPPublicKeyRing@5eca9ee0>, :secret #<org.bouncycastle.openpgp.PGPSecretKeyRing@364fc6ea>}

=> (def keyring *1)
#'user/keyring

=> (keyring/list-secret-keys (:secret keyring))
(#<org.bouncycastle.openpgp.PGPSecretKey@641eaee2>)

=> (def secret-key (first *1))
#'user/secret-key

```

I'll take a loot at BouncyCastle to see if I can figure out the right way to do this. The issue is that `PGPSecretKey` doesn't have an implementation of `Encodable`, `PGPPrivateKey` encodes as raw bytes into something that isn't decodable, and both need a matching `decode-*-key` function.

**Comment by greglook** [comment]
Well after looking around at the BC library for a bit, I think the issue stems from treating private keys the wrong way. Public keys can be encoded and handled separately, but secret and private keys must be passed around as part of a keyring. I pushed some commits to develop that let you use `encode` on keyrings, will test some more and cut a new version.

**Comment by greglook** [comment]
Available in 0.9.0.
